File: go/0724/2.go
package main

import (
	"fmt"
)

type Hyuck struct {
	Name	string
	Desc	string
}

func (h *Hyuck)Edit(n string, d string) {
	h.Name	= n
	h.Desc	= d
}

func (h *Hyuck)Print() {
	fmt.Printf("My name is %v and %v.\n", h.Name, h.Desc)
}

var h = Hyuck{
	Name: "Woodchuck",
	Desc: "I wonder why Golang has this \"pointer == variable\" feature",
}

func main() {
	h.Print()
	h.Edit("Roy", "I want to know the answer of that, too")
	h.Print()
}

File: c/pointer/2.c
#include <stdio.h>

int main() {
	int a = 6;
	int b = 3;
	// int* const p = &a; constant value that is a pointer to an int value
	const int* p = &a; // pointer to an int value that is a constant
	p = &b;
	return 0;
}
